src/sched/lthread.c

```diff
@@ -637,47 +707,37 @@ int lthread_create_primitive(
 {
     struct lthread* lt;
 
-    // FIXME: Remove when we no longer have lthread / libc layering issues.
-    if (!libc.threaded && libc.threads_minus_1 >= 0)
-    {
-        for (FILE* f = *__ofl_lock(); f; f = f->next)
-            init_file_lock(f);
-        __ofl_unlock();
-        init_file_lock(__stdin_used);
-        init_file_lock(__stdout_used);
-        init_file_lock(__stderr_used);
-        libc.threaded = 1;
-    }
-
     if ((lt = oe_calloc(1, sizeof(struct lthread))) == NULL)
     {
         return -1;
     }
 
     // FIXME: Once lthread / pthread layering is fixed, just use the tls
-    // argument as gs base.  We can't do that now because _lthread_free
+    // argument as fs base.  We can't do that now because _lthread_free
     // attempts to unmap this area.
-    lt->itlssz = libc.tls_size;
-    if (libc.tls_size)
-    {
-        if ((lt->itls = (uint8_t*)enclave_mmap(
-                 0,
-                 lt->itlssz,
-                 0, /* map_fixed */
-                 PROT_READ | PROT_WRITE,
-                 1 /* zero_pages */)) == MAP_FAILED)
-        {
-            oe_free(lt);
-            return -1;
-        }
-        if (__init_utp(__copy_utls(lt, lt->itls, lt->itlssz), 0))
-        {
-            oe_free(lt);
-            return -1;
-        }
-    }
+    //Rm tls_size from lt 
+    // lt->itlssz = libc.tls_size;
+    lt->itls = tls;
+    // if (libc.tls_size)
```

Comment:
You can also remove most of the equivalent changes in `lthread_create`.  That function should only be called for non-userspace threads and so doesn't need to touch any libc state.